### ISSUE TITLE
pkgrepo.managed key_url: salt:// always use `base` env

### ISSUE THREAD

**Issue by JensRantil** [saltstack/salt#11474]
I have an SLS in a non-`base` environment that uses the `pkgrepo.managed` state with `key_url: salt://...`. The key file resides in that samt non-base environment. However, it seems that `aptpkg.py` (which `pkgrepo.py` uses) is still trying to fetch the key file from `base` environment.

Minion log:
```bash
$ sudo salt-minion -l debug
...
[INFO    ] Running state [deb http://my-server precise main] at time 11:44:47.103746
[INFO    ] Executing state pkgrepo.managed for deb http://my-server precise main
[DEBUG   ] Fetching file from saltenv 'base', ** attempting ** 'salt://my.gpg.key'
[ERROR   ] Unable to cache file 'salt://my.gpg.key' from saltenv 'base'.
[INFO    ] Executing command 'apt-key add ' in directory '/tmp'
[ERROR   ] Command 'apt-key add ' failed with return code: 2
[ERROR   ] stderr: gpg: can't open `': No such file or directory
[ERROR   ] Failed to configure repo 'deb http://my-server precise main': Error: key retrieval failed: apt-key add
[INFO    ] Completed state [deb http://my-server precise main] at time 11:44:47.441661
...
```

Can anyone reproduce this?

For now, I'll use `file.managed` and `cmd.wait` as a workaround for this.

**Comment by amontalban** [comment]
Hi,

I just came across this issue Googling for what values were valid for `key_url` parameter as I wanted to use a file in Salt master. I can confirm that the original issue is not happening to me using (I'm using a staging environment instead of default base one).

My versions report:

```
root@ip-10-0-1-5:~# salt --versions-report
                  Salt: 2015.5.2
                Python: 2.7.3 (default, Mar 13 2014, 11:03:55)
                Jinja2: 2.6
              M2Crypto: 0.21.1
        msgpack-python: 0.1.10
          msgpack-pure: Not Installed
              pycrypto: 2.6
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 13.1.0
                  RAET: Not Installed
                   ZMQ: 3.2.3
                  Mako: 0.7.0
 Debian source package: 2015.5.2+ds-1~bpo70+1
```
It would be great if you can update the docs to say that `key_url` accepts a *salt://* path because URL sounds it is only *http://* or *https://*.

Thanks!

**Comment by rallytime** [comment]
@amontalban Just to clarify, you're saying that if you use `key_url` instead of `url` this is fixed, but we need to update the docs to reflect this change. Do I have that correct?

**Comment by amontalban** [comment]
@rallytime Sorry, I think I confused you guys. I have tested the following:

```
# Configure deb-multimedia repository for FFMPEG packages
deb-multimedia:
  pkgrepo.managed:
    - humanname: deb-multimedia
    - name: deb http://www.deb-multimedia.org {{ salt['grains.get']('oscodename') }} main
    - file: /etc/apt/sources.list.d/deb-multimedia.list
    - key_url: salt://deb-multimedia/files/marillat.pub
```
In the following version:
```
root@ip-10-0-1-5:~# salt --versions-report
                  Salt: 2015.5.2
                Python: 2.7.3 (default, Mar 13 2014, 11:03:55)
                Jinja2: 2.6
              M2Crypto: 0.21.1
        msgpack-python: 0.1.10
          msgpack-pure: Not Installed
              pycrypto: 2.6
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 13.1.0
                  RAET: Not Installed
                   ZMQ: 3.2.3
                  Mako: 0.7.0
 Debian source package: 2015.5.2+ds-1~bpo70+1
```
**It does work properly when using non-base environment** (staging in my case) so I guess the original issue is probably fixed in 2015.5. Regarding the documentation point I just wanted to let you know that would be great if you can add a note in the docs that says that `key_url` allows `http://` `https://` and `salt://` as I didn't know it allows `salt://` to use files, just and improvement to the docs.

Thanks!

**Comment by rallytime** [comment]
@amontalban Oh that makes more sense. I was just reading your original statement incorrectly. Thank you very much for the clarification and for confirming this is fixed. I've submitted #25831 to clarify that all three of those options are viable for `key_url` in the docs, which I think should finish up any remaining confusion.
